source/slang/lower-to-ir.cpp

```diff
@@ -1666,23 +1787,33 @@ struct ExprLoweringVisitorBase : ExprVisitor<Derived, LoweredValInfo>
 
         UInt argCount = expr->args.Count();
 
+        // If the initializer list was empty, then the user was
+        // asking for default initialization, which should apply
+        // to (almost) any type.
+        //
+        if(argCount == 0)
```

Comment:
Perhaps this should handle enums (and perhaps other types without defaults) without hitting SLANG_UNEXPECTED("unexpected type when creating default value");

Reply:
We could probably handle `enum` types by plugging in a zero value of their underlying "tag" type, but that isn't really your point. Yes, I agree that it would be good to provide a better behavior here for types that don't have a meaningful default, but that would really need to be a front-end check and I'm not yet sure what is the right way to go about that for the kind of language Slang is trying to be (in the case of a simple C-like language without separate compilation it is pretty easy to define a "can this be defaulted?" function).

I suppose for now the front-end can/should use a check similar to what the IR generation step is using to rule out the types we don't want to allow.